feat: add 'occupancies.txt' file (#38)
hove-io/ntfs-specification: ntfs_fr.md

[`pathways.txt`](#pathwaystxt-optionnel) | Optionnel | Ce fichier contient les cheminements au sein d'une zone d'arrêt. Ces cheminements ne sont pas nécessairement géographiques, il peut y avoir des simplifications.
[`levels.txt`](#levelstxt-optionnel) | Optionnel | Ce fichier contient la liste des niveaux au sein d'une zone d'arrêt.
[`addresses.txt`](#addressestxt-optionnel) | Optionnel | Ce fichier contient la liste des adresses des arrêts physiques.

> [`occupancies.txt`](#occupanciestxt-optionnel) | Optionnel | Ce fichier contient les informations d'affluence sur le réseau [**expérimental**]

Reviewer: ```suggestion
[`occupancies.txt`](#occupanciestxt-optionnel-expérimental) | Optionnel | Ce fichier contient les informations d'affluence sur le réseau [**expérimental**]
```

Author: Fixed in 0231f49. Thanks for catching that.

street_name | chaine | Requis | Nom de la voierie
house_number | chaine | Optionnel | Numéro du seuil

### occupancies.txt (optionnel)
Pour chaque ligne de ce fichier, l'affluence est appliquée à toutes les séries d'horaires successifs d'une circulation remplissant l'ensemble des conditions suivantes :

* les poteaux d'arrêts des horaires se trouvent entre `from_stop_area` et `to_stop_area` (inclus)

> * le jour de circulation est inclus entre `from_date` et `to_date`

Reviewer 2: ```suggestion
* le jour de circulation est inclus entre `from_date` et `to_date` (inclus)
```

Author: Done in 99c9dd7.

> Pour chaque ligne de ce fichier, l'affluence est appliquée à toutes les séries d'horaires successifs d'une circulation remplissant l'ensemble des conditions suivantes :
> 
> * les poteaux d'arrêts des horaires se trouvent entre `from_stop_area` et `to_stop_area` (inclus)
> * le jour de circulation est inclus entre `from_date` et `to_date`
> * le jour de semaine de la circulation est actif (voir champs `monday`, `tuesday`, etc.)
> * les `departure_time` des horaires sont inclus entre `from_time` et `to_time` (inclus)
> * il est possible de monter dans le véhicule sur le premier des horaires
> * il est possible de descendre dans le véhicule sur le dernier des horaires

Reviewer 3: Proposition qui me semble plus claire (qui au moins reformule ce que j'ai compris) :
```suggestion
La notion d'affluence est appliquée sur l'intervalle entre 2 horaires (affluence à bord du véhicule).
Pour chaque ligne de ce fichier, l'affluence est appliquée à tous les intervalles d'une circulation entre deux horaires (successifs ou non) remplissant l'ensemble des conditions suivantes :

* le premier des intervalles débute à l'arrêt `from_stop_area` et le dernier des intervalles se termine l'arrêt `to_stop_area` (inclus)
* le jour de circulation est inclus entre `from_date` et `to_date` (inclus)
* le jour de semaine de la circulation est actif (voir champs `monday`, `tuesday`, etc.)
* le `departure_time` du premier intervalle et le `arrival_time` du derniers intervalle sont inclus entre `from_time` et `to_time` (inclus)
* il est possible de monter dans le véhicule à l'arrêt `from_stop_area`
* il est possible de descendre du véhicule à l'arrêt `to_stop_area`
```

Reviewer 4: Nous avons eu une longue discussion avec @woshilapin sur le sujet. J'aime bien ta proposition de reformulation @pbougue mais si cette proposition fiabilise l'info pour éviter une information parfois fausse en cas de point de passage, on va se heurter à un soucis de prédictabilité de l'info sur des enrichissements côté Data.
J'ai proposé de retirer les critères de MI/DI pour appliquer systématiquement la règle entre le `from_stom_area` et le `to_stop_area`, mais les effets de bords  sont-ils acceptables ?
A suivre :christmas_tree: :santa:

Author: Correction of the text with your proposition done in 99c9dd7.

Reviewer 3: Je vous laisse voir s'il faut relancer la discussion de fiabilité vs. MDI (autre thread car ne concerne qu'une sous-partie de ce texte ?).
Je laisse donc ce thread ouvert, mais pour ce qui me concerne, c'est OK de le 'resolve'.

Author: J'ai assez envie de commencer l'implémentation pour avoir du _feedback_ concret à ce stade.

Reviewer 4: Je pense que cette version est good enough

> * il est possible de descendre dans le véhicule sur le dernier des deux horaires successifs

Reviewer 3: ```suggestion
* il est possible de descendre du véhicule sur le dernier des deux horaires successifs
```

Author: Done in 99c9dd7.

> Les lignes du fichier sont appliquées dans l'ordre.

Reviewer 3: Est-ce qu'il serait facile de gérer que le plus prioritaire est celui qui désigne exactement les 2 arrêts d'une section ?

Reviewer 4: Je pense que justement, l'idée est de pouvoir appliquer des règles génériques en premier, et de les surcharger à la suite du fichier.
Cette gestion par "règle" ressemble à ce qui est fait dans les tarifs GTFS, et c'est assez efficace à mon avis.

Author: On va conserver la proposition actuelle sur ce point ! Je résous le commentaire.

> 

Reviewer 3: Quid du cas : A - B - C - D, affluence fournie seulement pour la section A - D.
Est-ce que l'affluence remonte sur B - C, ou uniquement sur A - B et C - D ?
J'en comprends que les 3 sections auront l'affluence, il y a peut-être moyen d'être encore plus explicite, sinon pas de souci à laisser comme ça.

Author: Non, elle remonte sur toutes les successions de `stop_times` inclus, donc B-C également. Et c'est voulu à ce stade. Tu penses à un cas que j'aurais oublié ?

Reviewer 3: Non, je ne pense pas à un cas oublié, ici.

J'ai du mal à comprendre l'implication/l'usage de vérifier qu'on peut monter/descendre dans la section, je me demande si ça peut pas empêcher de remonter une info utile sur A - B - C "juste" parce que B est sans montée ni descente.

Author: Tu mets le doigt sur un cas intéressant... qui est couvert dans ma tête, mais effectivement mal retranscrit dans les spécifications.

- A-C spécifié dans le fichier
- un `trip` qui fait A-B-C mais avec interdiction de montée/descente en B

Dans ma tête, on remplit bien A-B et B-C. Mais avec l'état actuel de la spécification, ce n'est pas le cas. Je pense que mon erreur est de parler de deux horaires successifs... En réalité, c'est toutes successions d'horaire pour lesquels on trouve un début avec montée autorisée et une fin avec descente autorisée et qui est incluse dans les fourchettes d'espace et de temps. Je vais voir pour essayer de corriger.

Author: @pbougue @prhod Let me know if it improves or not? (see 4538243).

Reviewer 3: C'est mieux, ça me convient de laisser comme ça, même si j'ai fait une contre-proposition (n'hésite pas à resolve ici si tu veux déplacer la conversation).

street_name | chaine | Requis | Nom de la voierie
house_number | chaine | Optionnel | Numéro du seuil

### occupancies.txt (optionnel)
La notion d'affluence est appliquée sur l'intervalle entre 2 horaires (affluence à bord du véhicule).
Pour chaque ligne de ce fichier, l'affluence est appliquée à tous les intervalles d'une circulation entre deux horaires (successifs ou non) remplissant l'ensemble des conditions suivantes :

* le premier des intervalles débute à l'arrêt `from_stop_area` et le dernier des intervalles se termine l'arrêt `to_stop_area` (inclus)
* le jour de circulation est inclus entre `from_date` et `to_date` (inclus)
* le jour de semaine de la circulation est actif (voir champs `monday`, `tuesday`, etc.)

> * le `departure_time` du premier intervalle et le `arrival_time` du dernier intervalle sont inclus entre `from_time` et `to_time` (inclus)

Reviewer 3: Je déplace la discussion ici pour le confort des réponses.

@woshilapin à écrit:
> @pbench @prhod @pbougue 
> Since `occupancies.txt` has `from_date,to_date` and all the days of the week, would it be wise to reuse the existing `calendar.txt` instead? It doesn't mean we have to use existing calendar, but we could define new ones explicitly for occupancies (and I'm betting sometimes, we're going to reuse existing calendar)? That would simplify a great deal the current file and reuse existing objects.
>
> ```csv
> line_id,from_stop_area,to_stop_area,service_id,from_time,to_time,occupancy
> ```

De base, ça me semble plutôt bien de réutiliser les choses existantes et de mutualiser.
Je vois un souci à ça : le calendar est assez lié (un peu abusivement) aux trips (l'id s'appelle `service_id`).
Je vois un contrepoids (contournement possible ?) à ce souci : le grid_rel_calendar_line qui associe la notion de grid_calendar aux lines, et donc on pourrait envisager le même genre de choses pour l'affluence ?.

Après, je ne suis pas pointu sur la nuance grid-calendar vs. calendar... Mais rendre l'objet calendar multi-purpose me semble être une bonne direction dans la modélisation de plannings :smile_cat: (je ne sais pas s'il y a une meilleure façon de le faire avec des tables de relation-composition, avec rien, avec des "héritages").
Je pense qu'on a un ex-archi-data qui pourra avoir un avis éclairant sur ça !

Reviewer 4: Pour le moment, le modèle sur les taux d'occupation est expérimental, et les calendriers ne seront pas mutualisés. Il y a un risque de "nettoyage" des calendriers considérés comme inutilisés alors qu'ils sont référencés par l'occupation.
Je propose qu'on laisse les deux listes décorellées pour le moment (en indiquant le côté "experimental").

Author: Après discussion avec @prhod, c'est pousser l'optimisation un peu loin à ce stade de prototype. On se propose plutôt d'annoter la section entière avec un **Expérimental** histoire que l'API ne soit pas trop figée pour le moment. Il sera toujours temps d'y revenir plus tard.
Effectué dans https://github.com/hove-io/ntfs-specification/pull/38/commits/1fef488f62c2ae421afe0056644b5511e492def6.

Author: Je résous également cette discussion, car cette PR n'avancera pas plus sur ce sujet comme mentionné précédemment.
